api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_highly_variable_genes.py

```diff
@@ -0,0 +1,338 @@
+        highly_variable_rank=pd.Series(median_ranked, index=var_df.index),
+        variances_norm=pd.Series(np.mean(norm_gene_vars, axis=0), index=var_df.index),
+    )
+
+    sorted_index = (
+        var_df[["highly_variable_rank", "highly_variable_nbatches"]]
+        .sort_values(
+            ["highly_variable_rank", "highly_variable_nbatches"],
+            ascending=[True, False],
+            na_position="last",
+        )
+        .index
+    )
+    var_df["highly_variable"] = False
+    var_df.loc[sorted_index[: int(n_top_genes)], "highly_variable"] = True
+    if batch_key is None:
+        var_df = var_df.drop(columns=["highly_variable_nbatches"])
+    return var_df
+
+
+def highly_variable_genes(
+    query: soma.ExperimentAxisQuery,
+    n_top_genes: int = 1000,
+    layer: str = "raw",
+    flavor: Literal["seurat_v3"] = "seurat_v3",
+    span: float = 0.3,
+    batch_key: Optional[str] = None,
+) -> pd.DataFrame:
+    """
+    Identify and annotate highly variable genes contained in the query results.
+    The API is modelled on ScanPy `scanpy.pp.highly_variable_genes` API, and
+    results returned will mimic ScanPy results. The only `flavor` available
+    is the Seurat V3 method, which assumes count data in the X layer.
+
+    Args:
+        query:
+            A SOMA query, specifying the obs/var selection over which genes are annotated.
+
+        n_top_genes:
+            Number of genes to rank.
+
+        layer:
+            X layer used, e.g., `raw`
+
+        flavor:
+            Method used to annotate genes. Must be `seurat_v3`
+
+        span:
+            For `seurat_v3` flavor, the fraction of obs/cells used to
+            estimate the loess variance model fit.
+
+        batch_key:
+            If specified, gene selection will be done by batch and combined.
+
+    Returns:
+        Pandas DataFrame containing annotations for all `var` values specified by the
+        `query` argument. Annotations are identical to those produced by
+        `scanpy.pp.highly_variable_genes`
+
+    Raises:
+        ValueError: if the flavor paramater is not `seurat_v3`.
+
+    Lifecycle:
+        experimental
+    """
+    if flavor != "seurat_v3":
+        raise ValueError('`flavor` must be "seurat_v3"')
+
+    return _highly_variable_genes_seurat_v3(
+        query,
+        n_top_genes=n_top_genes,
+        layer=layer,
+        span=span,
+        batch_key=batch_key,
+    )
+
+
+def get_highly_variable_genes(
+    census: soma.Collection,
+    organism: str,
+    measurement_name: str = "RNA",
+    X_name: str = "raw",
+    obs_value_filter: Optional[str] = None,
+    obs_coords: Optional[SparseDFCoord] = None,
+    var_value_filter: Optional[str] = None,
+    var_coords: Optional[SparseDFCoord] = None,
+    n_top_genes: int = 1000,
+    flavor: Literal["seurat_v3"] = "seurat_v3",
+    span: float = 0.3,
+    batch_key: Optional[str] = None,
+) -> pd.DataFrame:
```

Comment:
Would it be worth having this convenience wrapper return the `combined_df`, per the example in the PR summary?

Reply:
It could, but that would require the fetch of the entire var dataframe, and I did not want to presume that the user would want that extra overhead. 

Most use cases I know of do _not_ fetch the entire var.

I'll add an example to the docstrings to show that use case.